pkgs/development/tools/analysis/valgrind/default.nix

```diff
@@ -14,10 +14,14 @@ stdenv.mkDerivation rec {
 
   # Perl is needed for `cg_annotate'.
   # GDB is needed to provide a sane default for `--db-command'.
-  buildInputs = [ perl ] ++ stdenv.lib.optional (!stdenv.isDarwin) gdb;
+  buildInputs = [ perl gdb ]  ++ stdenv.lib.optional (stdenv.isDarwin) [ mig dsymutil ];
+
+  __impureHostDeps = stdenv.lib.optional (stdenv.isDarwin) [ mig dsymutil ];
```

Comment:
Unless I'm mistaken this should be defined in mig and dsymutil, eg. `__impureHostDeps = [ "/usr/bin/dsymutil" ];`